benches/bench1.rs

```diff
@@ -475,6 +475,8 @@ fn merge_default(c: &mut Criterion) {
     let mut data1 = vec![0; 1024];
     let mut data2 = vec![0; 800];
     let mut x = 0;
+
+    #[allow(clippy::explicit_counter_loop, clippy::unused_enumerate_index)]
```

Comment:
I'm not sure what this benchmark is about. I'm only guessing that calling enumerate and then ignoring the index is intentional. Same goes for the similar ones below.

Reply:
@mightyiam After merging this PR, I search for PRs/issues about clippy (and closed some), saw #618 and eventually ran `cargo clippy --all-targets` and got `clippy::unused_enumerate_index` as "unknown lint" (yet I found it online). _What am I missing here?_
And second thing, we don't run clippy with `--all-target` and I'm wondering why, I don't see any discussion about it.

Reply:
I think we run clippy four times because of a matrix value `features`. One of the is `--all-targets --all-features`.

What version toolchain did you get an error with and what is the error exactly, please?

Reply:
Ahh! [unused_enumerate_index](https://rust-lang.github.io/rust-clippy/master/index.html#/unused_enumerate_index) is added in 1.75 and I'm still on 1.74, my bad sorry! I'm so used on thinking I can't use recent features of Rust I did not thought it could be such a recent lint.
And another mistake of mine on matrix features. Ok, thanks!